src/plugins/data_views/server/fetcher/index_patterns_fetcher.ts

```diff
@@ -72,6 +73,9 @@ export class IndexPatternsFetcher {
     // if only one pattern, don't bother with validation. We let getFieldCapabilities fail if the single pattern is bad regardless
     if (patternList.length > 1 && !allowNoIndices) {
       patternListActive = await this.validatePatternListActive(patternList);
+      if (patternListActive.length === 0) {
+        throw new DataViewMissingIndices(patternList.join(','));
+      }
```

Comment:
I've been testing in Discover and without this PR, the field list was invalid when there were multiple patterns without data:
<img width="1448" alt="Bildschirmfoto 2023-03-08 um 08 29 47" src="https://user-images.githubusercontent.com/463851/223651813-98f053b1-4960-4911-acdf-1cf004609ea3.png">
Applying this change caused the following state in Discover (we could certainly catch that but the question is, there might be other parts of Kibana having issues)
![Bildschirmfoto 2023-03-08 um 08 37 34](https://user-images.githubusercontent.com/463851/223652079-75d11f97-c70b-4661-9907-c94d4a8b66c2.png)

So I'd suggest to not throw an index, but just return the information that there were no fields and indices found


```suggestion
      if (patternListActive.length === 0) {
        return { fields: [], indices: [] };
      }
```
The Discover renders correctly
![Bildschirmfoto 2023-03-08 um 08 40 27](https://user-images.githubusercontent.com/463851/223652300-12d9b3e2-08e0-4552-a4ce-79161326b565.png)

Reply:
@mattkime does it look good to you? Thx!